File: Python/Problem #5 [Medium]/main.py
def car(pair):
    def get_first(a, b):
        return a
    return pair(get_first)

def cdr(pair):
    def get_last(a, b):
        return b
    return pair(get_last)

def cons(a, b):
    def pair(f):
        return f(a, b)
    return pair

print(car(cons(3, 4)))
print(cdr(cons(3, 4)))

File: TypeScript/Problem #5 [Medium]/main.js
function cons(a, b) {
    function pair(f) {
        return f(a, b);
    }
    return pair;
}
var car = function (pair) {
    return pair(function (a, b) { return a; });
};
var cdr = function (pair) {
    return pair(function (a, b) { return b; });
};
console.log(car(cons(3, 4)));
console.log(cdr(cons(3, 4)));
